source/simulator/solver.cc

```diff
@@ -659,12 +659,20 @@ namespace aspect
         // linearized_stokes_variables has a different
         // layout than current_linearization_point, which also contains all the
         // other solution variables.
-        linearized_stokes_initial_guess.block (block_vel) = current_linearization_point.block (block_vel);
+        if (parameters.nonlinear_solver != NonlinearSolver::Newton_Stokes)
+          {
+            linearized_stokes_initial_guess.block (block_vel) = current_linearization_point.block (block_vel);
 
-        linearized_stokes_initial_guess.block (block_p) = current_linearization_point.block (block_p);
-        denormalize_pressure (this->last_pressure_normalization_adjustment,
-                              linearized_stokes_initial_guess,
-                              current_linearization_point);
+            linearized_stokes_initial_guess.block (block_p) = current_linearization_point.block (block_p);
+            denormalize_pressure (this->last_pressure_normalization_adjustment,
+                                  linearized_stokes_initial_guess,
+                                  current_linearization_point);
+          }
+        else
+          {
+            linearized_stokes_initial_guess.block (block_vel) = 0;
+            linearized_stokes_initial_guess.block (block_p) = 0;
```

Comment:
can you add a comment here that says something like "the Newton solver solves for *updates* to variables, for which we have no good initial guesses. therefore, initialize the solution vector for the solver to zero."?